src/components/Calculator.js

```diff
@@ -1,52 +1,49 @@
 
-export default class Calculator extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-  }
-
-  componentDidMount() {
-    this.setState({
-      total: null,
-      next: null,
-      operation: null,
-    });
-  }
-
-  buttonClick = (value) => {
-    const oper = calculate({ ...this.state }, value);
-    this.setState(oper);
-  }
-
-  render() {
-    const numPad = Array.from(Array(10).keys());
-    const digitPanel = ['.', ...numPad, 'AC', '+/-', '%'].reverse().map((digit) => {
-      if (digit === 0) {
-        return <Button classList="calculator-btn doubleWidth" appender={this.buttonClick} value={digit.toString()} key={digit} />;
-      }
-      return <Button classList="calculator-btn" appender={this.buttonClick} value={digit.toString()} key={digit} />;
-    });
-
-    const operatorPad = ['+', 'x', '-', '/', '='];
-    const operatorPanel = operatorPad.map((operator) => <Button classList="calculator-btn btn-operators" appender={this.buttonClick} value={operator} key={operator} />);
-    const { total, next, operation } = this.state;
-    return (
-      <div className="calculator">
-        <Screen total={total} next={next} operation={operation} />
-        <div className="App-wraper">
-          <div className="App-Num-Pad">
-            { digitPanel }
-          </div>
-          <div className="App-Op-Pad">
-            { operatorPanel }
-          </div>
+const Calculator = () => {
+  const [total, setTotal] = useState(null);
+  const [next, setNext] = useState(null);
+  const [operation, setOperation] = useState(null);
+
+  const getState = () => (
+    { total, next, operation }
+  );
+
+  const buttonClick = (value) => {
+    const oper = calculate(getState(), value);
+    const { total: sum, next: nextVal, operation: validOperation } = oper;
+    if (sum !== undefined) setTotal(sum);
+    if (nextVal !== undefined) setNext(nextVal);
+    if (validOperation !== undefined) setOperation(validOperation);
+  };
+
+  const numPad = Array.from(Array(10).keys());
+  const digitPanel = ['.', ...numPad, 'AC', '+/-', '%'].reverse().map((digit) => {
+    if (digit === 0) {
+      return <Button classList="calculator-btn doubleWidth" appender={buttonClick} value={digit.toString()} key={digit} />;
+    }
+    return <Button classList="calculator-btn" appender={buttonClick} value={digit.toString()} key={digit} />;
+  });
+
+  const operatorPad = ['+', 'x', '-', '/', '='];
+  const operatorPanel = operatorPad.map((operator) => <Button classList="calculator-btn btn-operators" appender={buttonClick} value={operator} key={operator} />);
+
+  return (
+    <div className="calculator">
+      <Screen total={total} next={next} operation={operation} />
+      <div className="App-wraper">
+        <div className="App-Num-Pad">
+          { digitPanel }
+        </div>
+        <div className="App-Op-Pad">
+          { operatorPanel }
         </div>
       </div>
-    );
-  }
-}
+    </div>
+  );
+};
+
```

Comment:
- [ ] [OPTIONAL] Great job with converting class based components to a functional component using hooks 👍. One thing is that division is not working. Let's fix it 😀🙌💡